js/data/buffer.js

```diff
@@ -83,7 +83,7 @@ Buffer.prototype.setVertexAttribPointers = function(gl, program) {
  * @param gl The WebGL context
  */
 Buffer.prototype.destroy = function(gl) {
-    if (this.buffer) {
+    if (gl && this.buffer) {
```

Comment:
When would `gl` be falsy here? Instead of adding a guard clause, can we eliminate that case?